processing/src/main/java/io/druid/segment/DimensionSelectorUtils.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import io.druid.java.util.common.IAE;
+import io.druid.query.filter.ValueMatcher;
+import io.druid.segment.data.IndexedInts;
+import io.druid.segment.filter.BooleanValueMatcher;
+
+import java.util.BitSet;
+import java.util.Objects;
+
+public final class DimensionSelectorUtils
+{
+
+  private DimensionSelectorUtils()
+  {
+  }
+
+  public static ValueMatcher makeRowBasedValueMatcher(
```

Comment:
I wanted to emphasize the difference between DimensionSelector.makeValueMatcher() (interface methods) and DimensionSelectorUtils.makeValueMatcher() ("default" implementations of that interface methods, to which interface implementations may delegate).

Renamed to "makeValueMatcherGeneric" and added some Javadoc